File: backend/src/main/java/com/lol/backend/dto/GamePreviewDto.java
package com.lol.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class GamePreviewDto {

    private String name;
    private String tag;
    private String puuid;

    private String champion;
    private String championIcon;
    private String championSplashArt;

    private int kills;
    private int deaths;
    private int assists;
    private boolean win;

    private int goldEarned;
    private int damage;
    private int visionScore;

    private double kda;
}

File: frontend/src/app/core/models/game-preview.dto.ts
export interface GamePreviewDto {
  name: string;
  tag: string;
  puuid: string;

  champion: string;
  championIcon: string;
  championSplashArt: string;

  kills: number;
  deaths: number;
  assists: number;
  win: boolean;

  goldEarned: number;
  damage: number;
  visionScore: number;

  kda: number;
}
